src/WebJobs.Extensions.DurableTask/Bindings/BindingHelper.cs

```diff
@@ -23,6 +23,11 @@ public BindingHelper(DurableTaskExtension config)
         public IAsyncCollector<StartOrchestrationArgs> CreateAsyncCollector(DurableClientAttribute clientAttribute)
         {
             IDurableOrchestrationClient client = this.config.GetClient(clientAttribute);
+            if (clientAttribute.DurableRequiresGrpc)
```

Comment:
I'm trying to remember when `CreateAsyncCollector` gets called, and whether it gets called early enough for us to correctly configure the gRPC protocol. I'm also wondering if this gets called at all for out-of-proc, or whether it's for in-proc only. Can you confirm these questions, and if there are no concerns, explain via code comments?

Reply:
You're right, it isn't called in this code path, even though I could have sworn I hit the breakpoint during testing. I moved the function metadata parsing for the client to GetClient(), which is later than the other trigger types but still early enough that we can defer TaskHubWorker init until after it is needed.

There is still a possibility for unexpected behavior if (somehow) a user app defines an app with two or more client functions that don't all define DurableRequiresGrpc, where the app might throw inconsistent error types depending on which function executes first, but this is already an unsupported scenario, and we warn against it explicitly in the Functions Python SDK.